include/slot.h

```diff
@@ -152,6 +154,31 @@ G_GNUC_WARN_UNUSED_RESULT;
 RaucSlot* r_slot_get_parent_root(RaucSlot *slot)
 G_GNUC_WARN_UNUSED_RESULT;
 
+/**
+ * Returns the the data subdirectory corresponding to the current or given
+ * hash/checksum.
+ *
+ * To allow cleanup of data that corresponds to a specific image hash, we use a
+ * subdirectory in <slot data directory>/hash-<hash value>.
+ *
+ * If the directory does not exist, it is created.
+ *
+ * @param slot slot to get the directory for
+ * @param checksum optional checksum to use instead of the current one
+ *
+ * @return name of the directory, must be freed.
+ */
+gchar *r_slot_get_checksum_data_directory(const RaucSlot *slot, const RaucChecksum *checksum, GError **error)
+G_GNUC_WARN_UNUSED_RESULT;
+
+/**
+ * Removes subdirectories which match <slot data directory>/hash-<hash value>,
+ * but don't correspond to the current slot hash/checksum.
+ *
+ * @param slot slot to get the directory for
+ */
+void r_slot_clean_data_directory(const RaucSlot *slot);
```

Comment:
For me, the function name does not match well with what it does.

Maybe something like `r_slot_cleanup_checksum_data_directories()`?

Reply:
My impression was that it should only clean older hash dirs since it explicitly skips other files/subdirectories.